src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerActivityPublisher.cs

```diff
@@ -0,0 +1,220 @@
+            _logger = logger;
+            _flushIntervalMs = flushIntervalMs;
+            _timer = new Timer(OnTimer, null, Timeout.Infinite, Timeout.Infinite);
+            _uniqueActivities = new HashSet<ContainerFunctionExecutionActivity>();
+            _flushInProgress = 0;
+        }
+
+        private void Start()
+        {
+            _logger.LogInformation($"Starting {nameof(LinuxContainerActivityPublisher)}");
+
+            // start the timer by setting the due time
+            SetTimerInterval(_flushIntervalMs);
+        }
+
+        private void OnStandbyOptionsChange()
+        {
+            _logger.LogInformation($"Triggering {nameof(OnStandbyOptionsChange)}");
+
+            if (!_standbyOptions.CurrentValue.InStandbyMode)
+            {
+                Start();
+            }
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            _logger.LogInformation($"Initializing {nameof(LinuxContainerActivityPublisher)}");
+
+            if (_standbyOptions.CurrentValue.InStandbyMode)
+            {
+                _logger.LogInformation($"Registering {nameof(_standbyOptionsOnChangeSubscription)}");
+                _standbyOptionsOnChangeSubscription = _standbyOptions.OnChange(o => OnStandbyOptionsChange());
+            }
+            else
+            {
+                Start();
+            }
+
+            return Task.CompletedTask;
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
+        {
+            _logger.LogInformation($"Stopping {nameof(LinuxContainerActivityPublisher)}");
+
+            // stop the timer if it has been started
+            _timer?.Change(Timeout.Infinite, Timeout.Infinite);
+
+            return Task.CompletedTask;
+        }
+
+        private async void OnTimer(object state)
+        {
+            await FlushFunctionExecutionActivities();
+            SetTimerInterval(_flushIntervalMs);
+        }
+
+        private async Task FlushFunctionExecutionActivities()
+        {
+            try
+            {
+                if (Interlocked.CompareExchange(ref _flushInProgress, 1, 0) == 0)
+                {
+                    try
+                    {
+                        var currentActivities = new List<ContainerFunctionExecutionActivity>();
+                        if (TryGetCurrentActivities(currentActivities))
+                        {
+                            if (_lastHeartBeatTime.AddMinutes(5) < DateTime.UtcNow)
+                            {
+                                _logger.LogDebug($"Current activities count = {currentActivities.Count}");
+                                _lastHeartBeatTime = DateTime.UtcNow;
+                            }
+
+                            if (currentActivities.Any())
+                            {
+                                _logger.LogInformation($"Flushing {currentActivities.Count} function activities");
```

Comment:
As I mentioned before, I think these should all be Debug logs. Users don't want to see these in their AI streams